src/stream_base.h

```diff
@@ -162,6 +162,7 @@ class StreamResource {
                       uv_buf_t* bufs,
                       size_t count,
                       uv_stream_t* send_handle) = 0;
+  virtual bool IsAsync();
```

Comment:
This feels odd … I think this would be a bit clearer if it was a set/get pair on the `WriteWrap` instance, defaulting to `true`, where `LibuvStreamWrap::DoWrite` sets the value?

I don’t think `StreamResource` is the right place for this…

Reply:
I think this might be more of a naming issue maybe? I do think it conceptually belongs on StreamResource because it's supposed to represent whether there's anything in the queue. Right now it's mostly for `LibuvStreamWrap` but it's not guaranteed that's the only sync write we'll have.

Reply:
I mean, yeah, if this stays here I think we could come up with a better name :)

But what we *use* it for is checking whether a particular write was happening asynchronously, right?

Reply:
I guess it has a dual purpose. Let me think about it a bit more.

Reply:
@apapirovski Maybe make it `HasWriteQueue()`?